src/resdk/collection_tables.py

```diff
@@ -10,235 +10,265 @@
+
+    @property
+    @lru_cache()
+    def exp(self) -> pd.DataFrame:
+        """Return expressions table as a pandas DataFarm object.
 
-        :return: modified datetime in UTC
+        Which type of expressions (TPM, CPM, FPKM, ...) get returned depends on how the
+        data was processed. The expression type can be checked in the returned table
+        attribute `attrs['exp_type']`:
+
+        .. code-block:: python
+
+            exp = tables.exp
+            print(exp.attrs['exp_type'])
+
+        :return: table of expressions
         """
-        if self._modified is None:
-            newest_modified = max(s.modified for s in self.collection.samples.iterate())
-            # transform into UTC so changing timezones won't effect cache
-            self._modified = newest_modified.astimezone(pytz.utc)
-        return self._modified
+        exp = self._load_fetch(EXP)
+        self.gene_ids = exp.columns.tolist()
+        return exp
 
+    @property
     @lru_cache()
-    def metadata(self) -> pd.DataFrame:
-        """Return samples metadata in a table format.
+    def rc(self) -> pd.DataFrame:
+        """Return expression counts table as a pandas DataFarm object.
 
-        :return: table of metadata
+        :return: table of counts
         """
-        cache = load_cache(self.cache_dir, self.collection.slug, META, self.modified)
-        if cache is not None:
-            self._meta = cache
+        rc = self._load_fetch(RC)
+        self.gene_ids = rc.columns.tolist()
+        return rc
+
+    @property
+    @lru_cache()
+    def id_to_symbol(self) -> Dict[str, str]:
+        """Map of source gene ids to symbols.
 
-        if self._meta is None:
-            self._meta = self._get_metadata()
+        This also gets fetched only once and then cached in memory and on disc.
+        :attr:`CollectionTables.exp` or :attr:`CollectionTables.rc` must be called
+        before this as the mapping is specific to just this data. Its intended use is
+        to rename table column labels from gene ids to symbols.
 
-        save_cache(
-            self._meta, self.cache_dir, self.collection.slug, META, self.modified
-        )
-        return self._meta
+        Example of use:
 
-    def _get_metadata(self) -> pd.DataFrame:
-        """Flatten samples JSON metadata into a table."""
-        meta = pd.json_normalize([ts.metadata for ts in self._table_samples])
-        meta = meta.set_index("sample_slug").sort_index()
-        return meta
+        .. code-block:: python
+
+            exp = exp.rename(columns=tables.id_to_symbol)
+
+        :return: dict with gene ids as keys and gene symbols as values
+        """
+        species = self._data[0].output["species"]
+        source = self._data[0].output["source"]
+
+        if not self.gene_ids:
+            raise ValueError("Expression data must be used before!")
+
+        return self._mapping(self.gene_ids, source, species)
+
+    @staticmethod
+    def clear_cache() -> None:
```

Comment:
This could (should?) be a function in `utils/table_cache.py`. Imagine we implement some other object, similar to `CollectionTables` that will also use cache... It would make sense that same this function goes along those that perform saving / loading cache.